src/Application/SaveMappings/SaveMappingsUseCase.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseRDF\Application\SaveMappings;
+
+use ProfessionalWiki\WikibaseRDF\Application\Mapping;
+use ProfessionalWiki\WikibaseRDF\Application\MappingList;
+use ProfessionalWiki\WikibaseRDF\Application\MappingRepository;
+use Throwable;
+use Wikibase\DataModel\Entity\EntityId;
+
+class SaveMappingsUseCase {
+
+	/**
+	 * @param string[] $allowedPredicates
+	 */
+	public function __construct(
+		private SaveMappingsPresenter $presenter,
+		private MappingRepository $repository,
+		private array $allowedPredicates
+	) {
+	}
+
+	public function saveMappings( EntityId $entityId, MappingList $mappings ): void {
+		$invalidMappings = $this->getInvalidMappings( $mappings );
```

Comment:
This does not currently handle invaliditiy due to the validation in `Mapping::__construct()`. If we want this use case to handle that as well, then we need to move the deserialization here. Otherwise, that (plus #31 and #32) will just need to create their responses directly in the Api class.

Reply:
Please see "Lesson learned: validation" in https://www.entropywins.wtf/blog/2016/11/24/implementing-the-clean-architecture/

Reply:
Current thoughts:

* Don't use MappingList and Mapping in the Request Model of the UC
* JSON validation happens in the presentation layer, including checks that we get strings and all required values are present
* Policy validation happens in the UC. For instance that only certain prefixes are allowed.
* Basic URL validation seems to best fit in the UC as well